quantum/rgb_matrix/rgb_matrix.c

```diff
@@ -250,7 +251,7 @@ void process_rgb_matrix(uint8_t row, uint8_t col, bool pressed) {
     if (pressed)
 #    endif // defined(RGB_MATRIX_KEYRELEASES)
     {
-        if (rgb_matrix_config.mode == RGB_MATRIX_TYPING_HEATMAP) {
+        if (g_heatmap_enabled || rgb_matrix_config.mode == RGB_MATRIX_TYPING_HEATMAP) {
```

Comment:
I completely agree with you!  Literally every other animation I have worked on has been able to tap-in well to existing functionality, allowing clean separation between quantun and kb levels.  This is the one place I wasn't able to do it!

The problem is I am not sure I am the right person for the job. I don't have enough experience with QMK to know the best way to do this, and cannot envision all the consequences.  I thought this PR would be a minimally invasive way to gain access to the heatmap functionality in a custom animation (until someone recognizes the suboptimal situation and does exactly as you suggest).

Reply:
Would you be interested/willing to collaborate on this?  I would be more comfortable messing about in the quantum realm with advise/oversight of someone more expert.

Reply:
Is it possible one might still want the heatmap tracking to be mode-enabled (with a flag), even if process_rgb_matrix_typing_heatmap is moved to rgb_matrix.c?  Perhaps on some MCUs it would be a performance issue, and so you'd want it disabled while other animations are active.